.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: CI Workflow
+on:
+  pull_request:
+    branches: [main]
+
+  push:
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
+
+      - name: Install wasm-pack
+        uses: jetli/wasm-pack-action@v0.4.0
+
+      - name: Cache WASM build
+        uses: actions/cache@v3
+        with:
+          path: rust_backend/target
+          key: doesnt-matter-share-everything
```

Comment:
Probably doesn't matter 100% but a more informative key here might be useful. I am assuming this invalidates the cache as well when rust_backend/target changes.

Reply:
I copied that line from ` .github/workflows/web.yml`. Happy to change to something else. Would that be applicable there as well?

Reply:
Changed to `rust_backend` in ad30590 as a demonstration @dabreegster. Totally fine to change etc.